controller/controller_server/csi_controller_server.py

```diff
@@ -534,6 +539,88 @@ def GetCapacity(self, request, context):
         logger.info("finished GetCapacity")
         return csi_pb2.GetCapacityResponse()
 
+    def ControllerExpandVolume(self, request, context):
+        set_current_thread_name(request.volume_id)
+        logger.info("ControllerExpandVolume")
+        secrets = request.secrets
+
+        try:
+            utils.validate_expand_volume_request(request)
+
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            try:
+                array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+            except ObjectIdError as ex:
+                logger.warning("volume id is invalid. error : {}".format(ex))
+                context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+                return csi_pb2.ControllerExpandVolumeResponse()
+
+            with get_agent(user, password, array_addresses, array_type).get_mediator() as array_mediator:
+                logger.debug(array_mediator)
+
+                required_bytes = request.capacity_range.required_bytes
+                min_size = array_mediator.minimal_volume_size_in_bytes
+                max_size = array_mediator.maximal_volume_size_in_bytes
+
+                volume_before_expand = array_mediator.get_object_by_id(volume_id, config.VOLUME_TYPE_NAME)
+                if not volume_before_expand:
+                    raise controller_errors.ObjectNotFoundError(volume_id)
+
+                if required_bytes == 0 or volume_before_expand.capacity_bytes >= required_bytes:
+                    context.set_code(grpc.StatusCode.OK)
+                    return utils.generate_csi_expand_volume_response(volume_before_expand.capacity_bytes,
+                                                                     node_expansion_required=False)
+
+                if not required_bytes <= max_size:
```

Comment:
IMO, `if required_bytes > max_size:` is more readable

Reply:
done